packages/email-petition/src/email-petition.tsx

```diff
@@ -183,16 +198,99 @@ export function EmailPetition(props: {
 
   const outsideUS = watch("outsideUS");
   const zip = watch("zip");
-  const isInSonomaCounty = useMemo(() => {
-    return zip && zip in SonomaCities;
-  }, [zip]);
-  const cities = useMemo(() => {
-    if (!isInSonomaCounty) {
-      return [];
+
+  // Fetch postcode_localities from Google Geocoding API
+  async function fetchPostcodeLocalities(zipcode: string) {
+    try {
+      const response = await ky
+        .get(
+          `https://maps.googleapis.com/maps/api/geocode/json?address=${zipcode}&key=${props.areaScope.googleGeocodingApiKey}`,
+        )
+        .json<{ results: any[] }>();
+
+      const scopeComponentType = (() => {
+        switch (props.areaScope.subdivision) {
+          case "city":
+            return "locality";
+          case "county":
+            return "administrative_area_level_2";
+          case "state":
+            return "administrative_area_level_1";
+          case "country":
+            return "country";
+        }
+      })();
+
+      const scopeName = props.areaScope.name;
+      const scopeNameLower = scopeName.toLowerCase();
```

Comment:
no need for "scopeName" variable if it's only going to be used once and the name doesn't clarify anything. you can just do:

```suggestion
      const scopeNameLower = props.areaScope.name.toLowerCase();
```